src/deploy/functions/release/reporter.ts

```diff
@@ -29,8 +39,185 @@ export class DeploymentError extends Error {
+      reports.push(
+        track.track(
+          "functions_deploy_result",
+          "partial_error_ratio",
+          totalErrors / (totalSuccesses + totalErrors)
+        )
+      );
+    } else {
+      reports.push(track.track("functions_deploy_result", "failure", totalErrors));
+    }
+  }
+
+  await utils.allSettled(reports);
+}
+
+/** Print error messages for failures in summary. */
+export function printErrors(summary: Summary): void {
+  const errored = summary.results.filter((r) => r.error) as Array<Required<DeployResult>>;
+  if (errored.length === 0) {
+    return;
+  }
+
+  errored.sort((left, right) => backend.compareFunctions(left.endpoint, right.endpoint));
+  logger.info("");
+  logger.info(
+    "Functions deploy had errors with the following functions:" +
+      errored
+        .filter((r) => !(r.error instanceof AbortedDeploymentError))
+        .map((result) => `\n\t${getFunctionLabel(result.endpoint)}`)
+        .join("")
+  );
+
+  printIamErrors(errored);
+  printQuotaErrors(errored);
+  printAbortedErrors(errored);
+}
+
+/** Print errors for failures to set invoker. */
+function printIamErrors(results: Array<Required<DeployResult>>): void {
+  const iamFailures = results.filter(
+    (r) => r.error instanceof DeploymentError && r.error.op === "set invoker"
+  );
+  if (!iamFailures.length) {
+    return;
+  }
+
+  logger.info("");
+  logger.info(
+    "Unable to set the invoker for the IAM policy on the following functions:" +
+      iamFailures.map((result) => `\n\t${getFunctionLabel(result.endpoint)}`).join("")
+  );
+
+  logger.info("");
+  logger.info("Some common causes of this:");
+  logger.info("");
+  logger.info(
+    "- You may not have the roles/functions.admin IAM role. Note that " +
+      "roles/functions.developer does not allow you to change IAM policies."
+  );
+  logger.info("");
+  logger.info("- An organization policy that restricts Network Access on your project.");
+
+  // We implicitly set IAM permissions to public invoker when creating a function that
+  // has no explicit invoker set. If these failures were on an inferred setInvoker command
+  // we need to let the customer know that it needs to be explicit next time.
+  const hadImplicitMakePublic = iamFailures.find(
+    (r) => backend.isHttpsTriggered(r.endpoint) && !r.endpoint.httpsTrigger.invoker
+  );
+  if (!hadImplicitMakePublic) {
+    return;
+  }
+  logger.info("");
+  logger.info(
+    "One or more functions were being implicitly made publicly available on function create."
+  );
+  logger.info(
+    "Functions are not implicitly made public on updates. To try to make " +
+      "these functions public on next deploy, configure these functions with " +
+      `${clc.bold("invoker")} set to ${clc.bold(`"public"`)}`
+  );
+}
+
+/** Print errors for failures with the GCF API. */
+function printQuotaErrors(results: Array<Required<DeployResult>>): void {
+  const hadGcfFailure = results.find(
```

Comment:
I'm having trouble connecting errors in the GCF API and with quota errors. Are we assuming that any deployment error in the GCF API is attributable to quota issue?